### PR TITLE
Remove Multidex usages

### PR DESCRIPTION
**Pull request by MGaetan89**
## Summary

Since the min SDK is 21, it is no longer necessary to use the Multidex library.

See the following for more info: https://developer.android.com/build/multidex#mdex-on-l

## Safety Assurance

- [ ] If the PR is high risk, "High Risk" label is set
- [ ] I have confidence that this PR will not introduce a regression for the reasons below
- [ ] Do we need to enhance manual QA test coverage ? If yes, "QA Note" label is set correctly

### CODE REVIEW

**Comment by damagatchi** [comment]
Can one of the admins verify this patch?

File: README.md [hunk 1]
```diff
@@ -40,7 +40,6 @@ git clone https://github.com/dimagi/commcare-core.git
 - Click "OK" to use the Gradle wrapper
 - Wait while Android Studio spins its wheels
 - Download any build dependencies that the SDK Manager tells you you need.
-- Disable _Instant Run_ found in Settings > Build, Execution, Deployment > Instant Run. (It does not play well with multidexing, which we have enabled, or with some of the processes we have set up for Google Services)
```
Comment (MGaetan89) [inline README.md:43]: That menu doesn't seem to exist anymore, so I took the liberty to remove the whole line. I can just remove the part about Multidex if you prefer.

**Comment by MGaetan89** [comment]
I see that the Linter check is failing. If I run the `google-java-format` (version 1.24.0) locally on `CommCareApplication`, the whole file gets reformatted. So I'm not sure if I should use an other version of `google-java-format` or if there is another issue to fix?

**Comment by shubham1g5** [comment]
@damagatchi ok to test

**Comment by shubham1g5** [comment]
@MGaetan89 Unfortunately our lint check checks for the complete files instead of the lines you changed. I see that there are no failures corresponding to your changes so it's fine to ignore the lint failure at the moment. Thanks!

**Comment by shubham1g5** [comment]
@damagatchi retest this please

**Comment by shubham1g5** [comment]
@damagatchi retest this please

**Comment by shubham1g5** [comment]
@damagatchi retest this please

**Comment by shubham1g5** [comment]
Facing some issues with memory config on Jenkins, retuned and retrying build. 
@damagatchi retest this please

**Comment by shubham1g5** [comment]
@damagatchi retest this please

**Review by shubham1g5 — APPROVED** [review]
Thanks for getting this in @MGaetan89!

**Pull request merged**